samza-core/src/main/java/org/apache/samza/runtime/AbstractApplicationRunner.java

```diff
@@ -26,7 +26,7 @@
 
 public abstract class AbstractApplicationRunner implements ApplicationRunner {
```

Comment:
> Clearly, I didn't review the PR that committed this changed. So sorry about asking questions this late
Yes you did: https://github.com/apache/samza/pull/74

> So, I am trying to understand the motivation behind this choice.
There were a few reasons, but the primary one is that if the streamFromConfig() method were implemented as a default in the interface, it would require the Config as a parameter and since all the Runners take config as a constructor parameter, this looks clunky to the user. It's better to have them provide the config only once.